File: OOP Using Java/BinarySearch.java
import java.io.*;
import java.util.Scanner;

class BinarySearch{
	static int flag = 0;
	
	public static void binarySearch(int arr[], int ub, int search){
		int l = 0, u = ub;
		
		while (l <= u){
			int mid = (l+u)/2;
			
			if (arr[mid] == search){
				System.out.println("The Search Element Has Been Found At Position "+ (mid+1));
				flag = 1;
				break;
			}
			else if (arr[mid] > search)
				u = mid - 1;
			else
				l = mid + 1;
		}
		if (flag == 0)
			System.out.println("The Search Element Has Not Been Found");
	}
	
	public static void main(String args[]){
		BinarySearch obj = new BinarySearch();
		Scanner s = new Scanner(System.in);
		int[] arr = new int[20];
		int i;
		
		System.out.print("Enter The No. Of Elements: ");
		int n = s.nextInt();
		
		for (i = 0; i < n; i++){
			System.out.print("Enter The Element " + (i+1) + " : ");
			arr[i] = s.nextInt();
		}
		
		System.out.print("Enter The Search Element: ");
		int search = s.nextInt();
		obj.binarySearch(arr,i,search);
	}
}

File: Cycle 4/Searching Algorithms/binarySearch.c
#include <stdio.h>

int arr[50], n, search;

void display(int arr[]){
    printf("{");
    for (int i = 0; i < n; i++)
        printf("\t%d", arr[i]);
    printf("\t}\n");
}

void initArr(){
    printf("Enter The Number Of Elements: ");
    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        printf("Enter element %d: ", i + 1);
        scanf("%d", &arr[i]);
    }

    printf("Created Array: ");
    display(arr);
}

void bubbleSort(int arr[]){
    int i, j, swapped;
    for (i = 0; i < n - 1; i++) {
        swapped = 0;
        for (j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swapped = 1;
            }
        }
        if (swapped == 0)
            break;
    }
}

void binarySearch(int arr[], int search){
    int l = 0, u = n - 1, mid, flag = 0;

    while (l <= u){
        mid = (l + u) / 2;
        if (arr[mid] == search){
            printf("Element Found At Position %d\n", (mid + 1));
            flag = 1;
            break;
        }
        else if (arr[mid] < search)
            l = mid + 1;
        else
            u = mid - 1;
    }
    if (flag == 0)
        printf("The Element Is Not Present In The Array\n");
}

int main(){
    initArr();
    
    bubbleSort(arr);
    printf("Sorted Array: ");
    display(arr);
    
    printf("Enter The Element To Search For: ");
    scanf("%d", &search);
    
    binarySearch(arr, search);
    return 0;
}
